### PR TITLE
Add support for sqlite

### LINKED ISSUE #423: Support sqlite as DB_DIALECT

### PR DESCRIPTION
**Pull request by rokerzfirst101**
**Description:**
- Added `sqlite3` as an acceptable value for `DB_DIALECT`
- Add import for `github.com/mattn/go-sqlite3` driver
- Updated the logic to initialize the connection string using only`DB_HOST` for SQLite
- The server/app needs to be built with `CGO_ENABLED=1` in order for the SQLite driver to work

**How it works:**

To use SQLite as the Database, the user would need to provide the following configs or environment variables:
```env
DB_DIALECT=sqlite3
DB_HOST={connection-string}
```

Connection String Ref: https://github.com/mattn/go-sqlite3?tab=readme-ov-file#connection-string

Closes #423

### CODE REVIEW

**Comment by Umang01-hash** [comment]
@rokerzfirst101 Can you please also update the Test function `TestSQL_getDBConnectionString`and add the test case for newly added code.

**Comment by rokerzfirst101** [comment]
sure

**Comment by vipul-rawat** [comment]
@rokerzfirst101 - please add/extend documentation for using Sqlite in gofr - https://gofr.dev/docs/advanced-guide/dealing-with-datasources

**Comment by rokerzfirst101** [comment]
@vipul-rawat I've updated the database documentation

**Comment by vikash** [comment]
Adding a requirement for cgo reduces compatibility. Many people do prefer to use go without cgo. There is an implementation which does not require cgo, we should prefer that.

**Review by vikash — CHANGES_REQUESTED** [review]
Adding a requirement for cgo reduces compatibility. Many people do prefer to use go without cgo. There is an implementation which does not require cgo, we should prefer that

File: pkg/gofr/migration/sql.go [hunk 1]
```diff
@@ -21,8 +22,6 @@ const (
     constraint primary_key primary key (version, method)
 );`
 
-	checkSQLGoFrMigrationsTable = `SELECT EXISTS (SELECT 1 FROM information_schema.tables WHERE table_name = 'gofr_migrations');`
-
```
Comment (rokerzfirst101) [inline pkg/gofr/migration/sql.go:25]: this statement will not work when there are multiple databases with the `gofr_migrations` table

**Comment by rokerzfirst101** [comment]
I've replaced the `github.com/mattn/go-sqlite3` pkg with `modernc.org/sqlite`

**Comment by aryanmehrotra** [comment]
@rokerzfirst101 Rather than taking connection string from the user, we can take the configs in the following way and it keep consistent across for all SQL databases which we currently support
```
DB_HOST=localhost #Not possible to add? does it support connecting to a remote file in that case it can be supported
DB_USER=root # can be used
DB_PASSWORD=password # can be used
DB_NAME=test #db_name?
DB_DIALECT=mysql #sqlite will be used here
```

**Comment by rokerzfirst101** [comment]
closing as stale, will add another one

**Pull request closed without merging**